src/contracts/AbstractARM.sol

```diff
@@ -432,17 +451,17 @@ abstract contract AbstractARM is OwnableOperable, ERC20Upgradeable {
         IERC20(liquidityAsset).transferFrom(msg.sender, address(this), assets);
 
         // mint shares
-        _mint(msg.sender, shares);
+        _mint(liquidityProvider, shares);
 
         // Add the deposited assets to the last available assets
         lastAvailableAssets += SafeCast.toInt128(SafeCast.toInt256(assets));
 
         // Check the liquidity provider caps after the new assets have been deposited
         if (capManager != address(0)) {
```

Comment:
I've just realised we can avoid the extra call to capManager if we also check the msg.sender is Zapper.
I'll change no the `nicka/lp` branch as this has already been merged